### PR TITLE
[MM-27919] Integrate end to end testing with the server

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR adds the necessary changes to enable end-to-end testing in `mmctl`. Instead of using a "detached" approach in which it needs the server to be running and there is a single state that persist between tests, it reuses the `api4` test helper to create and drop isolated databases to run the tests in.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28093

### CODE REVIEW

File: commands/mmctl_test.go [hunk 1]
```diff
@@ -36,20 +36,61 @@ func (s *MmctlUnitTestSuite) TearDownTest() {
...
+	})
+
+	s.Run(testName+"/LocalClient", func() {
+		fn(s.th.LocalClient)
+	})
+}
+
+// RunForAllClients runs a test function for all the clients
+// registered in the TestHelper
+func (s *MmctlE2ETestSuite) RunForAllClients(testName string, fn func(client.Client)) {
```
Comment (wiggin77) [inline commands/mmctl_test.go:84]: Nice way to handle this.

File: commands/user_e2e_test.go [hunk 2]
```diff
@@ -7,26 +7,68 @@ import (
...
+		_, appErr := s.th.App.UpdateActive(user, false)
+		s.Require().Nil(appErr)
+
+		err := userActivateCmdF(c, &cobra.Command{}, []string{user.Email})
+		s.Require().NoError(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Activate user wis.thout permissions", func() {
```
Comment (wiggin77) [inline commands/user_e2e_test.go:32]: typo

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:    One minor typo.

File: commands/user_e2e_test.go (hunk 2 above)
Comment (mgdelacroix) [reply commands/user_e2e_test.go:32]: oops, thanks! updating...

**Comment by mgdelacroix** [comment]
Apologies @wiggin77, requesting another review as I forgot to update the README section with the new test instructions. PTAL

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

**Review by wiggin77 — APPROVED** [review]
Good docs.  :+1:

**Review by lindy65 — APPROVED** [review]
Thanks @mgdelacroix 👍

**Pull request merged**